core/src/main/java/org/ldaptive/transport/DefaultOperationHandle.java

```diff
@@ -111,6 +147,24 @@ public class DefaultOperationHandle<Q extends Request, S extends Result> impleme
    * @param  timeout  duration to wait for a response
    */
   public DefaultOperationHandle(final Q req, final TransportConnection conn, final Duration timeout)
+  {
+    this(req, conn, timeout, DEFAULT_RESPONSE_TIMEOUT_CONDITION);
+  }
+
+
+  /**
+   * Creates a new operation handle.
+   *
+   * @param  req  request to expect a response for
+   * @param  conn  the request will be executed on
+   * @param  timeout  duration to wait for a response
+   * @param  timeoutCondition  whether a message should reset the timeout
+   */
+  DefaultOperationHandle(
+    final Q req,
+    final TransportConnection conn,
+    final Duration timeout,
+    final Predicate<Message> timeoutCondition)
```

Comment:
I can't imagine any value to exposing this condition to your API, and yet I see costs. Moving the signaling out of `complete` into two distinct methods, `completeMessage` and `exception`, is harder to follow and _feels_ riskier. You want to guarantee that the semaphore signals on both errors and responses, and the more you disperse that signaling the harder it is to reason through the concurrency as well as ensure that it actually happens. I'm not sure you've gained enough for that increase in complexity/risk.

Reply:
If you truly need two different predicates, a default and a distinct one for search, then you'd be better served in my opinion by using a protected abstract template method that returns the correct value based on context. In that way it would not be exposed to callers as an API input.

Reply:
I tend to agree. I was leaving myself some rope in case we wanted to support the old behavior for searches. I'll push a fix that uses a `protected` method.